sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -170,20 +170,23 @@ namespace Azure { namespace Core { namespace Http {
     }
   };
 
-#ifdef AZ_PLATFORM_POSIX
   /**
    * @brief #BodyStream providing its data from a file.
    */
   class FileBodyStream : public BodyStream {
   private:
-    // in mutable
+#if defined(AZ_PLATFORM_POSIX)
     int m_fd;
+#elif defined(AZ_PLATFORM_WINDOWS)
+    HANDLE m_hFile;
+#endif
     int64_t m_baseOffset;
     int64_t m_length;
     // mutable
     int64_t m_offset;
 
   public:
+#if defined(AZ_PLATFORM_POSIX)
```

Comment:
What API would end up showing in the generated doxygen docs that we publish? Ideally, we'd want both to show up so users know, and not hide one just because we happen to generate it on a Windows CI machine before publishing.

Reply:
I am not sure how we can make that possible.
We might need to ask EngSys for advice.
Our pipelines will generate Windows docs only. Even with our previous approach, the Linux only documentation won't show up.
I wouldn't care too much about it anyway.

Reply:
Maybe we can do in-class
```
typedef
#ifdef AZ_PLATFORM_WINDOWS
    HANDLE
#elif defined(AZ_PLATFORM_POSIX)
    int
#endif
    FileHandle;
```

then document it saying "defined as `HANDLE` on Windows, `int` on POSIX", and then use it everywhere you use HANDLE/int depending on the platform (i.e. `FileHandle m_fileHandle`)

Reply:
That's not a bad idea. Definitely worth trying and see how the generated docs look like.

Reply:
@antkmsft sounds like a good idea.
My only question is, as an end user, when I want to use the FileBodyStream, do I need to know what FileHandle is?
If I am in Windows and try to create the FileBodyStream, do I need to cast/convert my HANDLE to FileHandle?  or is that seamless for me?
I wouldn't like to confuse the end-user when looking into intellisense about this new type just in favor of the documentation when the type signature for itself would provide better self documentation

Reply:
The alternative (which is where we are at today) is the correct API won't even show up in the docs for Linux.

Reply:
sort of.
As a linux user, I can always generate the Documentation locally, in such case, I would see the Linux only docs.
Also, as a Linux user, the intellisense docs that are shown to me from Read() are the Linux version.

So the problem is only for me, when I go to the official SDK API docs and look for the FileBodyStream. I would be looking at Windows version